idaes/models/unit_models/tests/test_equilibrium_reactor.py

```diff
@@ -242,9 +242,39 @@ def test_conservation(self, sapon):
         )
 
     @pytest.mark.ui
-    @pytest.mark.unit
+    @pytest.mark.component
     def test_report(self, sapon):
-        sapon.fs.unit.report()
+        stream = StringIO()
+
+        sapon.fs.unit.report(ostream=stream)
+
+        output = """
+====================================================================================
+Unit : fs.unit                                                             Time: 0.0
+------------------------------------------------------------------------------------
+    Unit Performance
+
+    Variables: 
+
+    Key             : Value  : Units  : Fixed : Bounds
+          Heat Duty : 0.0000 :   watt :  True : (None, None)
+    Pressure Change : 0.0000 : pascal :  True : (None, None)
+
+------------------------------------------------------------------------------------
+    Stream Table
+                                             Units            Inlet     Outlet  
+    Volumetric Flowrate                meter ** 3 / second  0.0010000  0.0010000
+    Molar Concentration H2O              mole / meter ** 3     55388.     55388.
+    Molar Concentration NaOH             mole / meter ** 3     100.00  0.0026894
+    Molar Concentration EthylAcetate     mole / meter ** 3     100.00  0.0026894
+    Molar Concentration SodiumAcetate    mole / meter ** 3     0.0000     99.997
+    Molar Concentration Ethanol          mole / meter ** 3     0.0000     99.997
+    Temperature                                     kelvin     303.15     304.32
+    Pressure                                        pascal 1.0132e+05 1.0132e+05
+====================================================================================
+"""
+
+        assert output in stream.getvalue()
```

Comment:
- Defining intermediate variables could help pytest generate more readable diffs in case the assertion fails
- If the only reason we're using `in` instead of `==` in the assertion is the possible presence of newlines in either of the strings, we could `.strip()` both of them before comparing (I'm not sure if this has an effect on the failed assertion message)

```suggestion
        expected = """
====================================================================================
Unit : fs.unit                                                             Time: 0.0
------------------------------------------------------------------------------------
    Unit Performance

    Variables: 

    Key             : Value  : Units  : Fixed : Bounds
          Heat Duty : 0.0000 :   watt :  True : (None, None)
    Pressure Change : 0.0000 : pascal :  True : (None, None)

------------------------------------------------------------------------------------
    Stream Table
                                             Units            Inlet     Outlet  
    Volumetric Flowrate                meter ** 3 / second  0.0010000  0.0010000
    Molar Concentration H2O              mole / meter ** 3     55388.     55388.
    Molar Concentration NaOH             mole / meter ** 3     100.00  0.0026894
    Molar Concentration EthylAcetate     mole / meter ** 3     100.00  0.0026894
    Molar Concentration SodiumAcetate    mole / meter ** 3     0.0000     99.997
    Molar Concentration Ethanol          mole / meter ** 3     0.0000     99.997
    Temperature                                     kelvin     303.15     304.32
    Pressure                                        pascal 1.0132e+05 1.0132e+05
====================================================================================
"""
        generated = str(stream.getvalue())
        assert generated.strip() == expected.strip()
```

Reply:
I'm running into a similar issue debugging output checks in #711 , and am wary of spacing differences inside the string between lines as well. If we mainly wish to test the content we could remove the newlines and spacing before comparing, such as `assert generated.replace(" ", "").replace("\n", "") == expected.replace(" ", "").replace("\n", "")`. There's probably a more aesthetic way of doing this but it would ensure any string comparison failures are due to incorrect values and not formatting.

Reply:
I went ahead and implemented the approach above in my tests for the PR linked above, and the tests are more stable for long outputs. Additionally, when there are no new lines or spaces the error report will actually skip identical leading or trailing characters as below, which greatly speeds up debugging and yields more helpful failure messages. In this case, a few of the values changed by tiny amounts which altered the final string output (e.g. -59.350**06318307154** should be corrected to -59.350**063183071555**):

```
>           assert output_string == stream_string
E           AssertionError: assert 'Extentofreac...=============' == 'Extentofreac...============='
E             Skipping 222 identical leading characters in diff, use -v to show
E             Skipping 2613 identical trailing characters in diff, use -v to show
E             - 063183071555DutyfromReaction(MW)):28.21686516526679Compressorwork(MW):-5.785653099883788e-25Turbinework(MW):-2.4215308912405678FeedMixeroutlettemperature(C)):20.051714213753655RecycleMixeroutlettemperature(C)):40.21858118534789FeedCompressoroutlettemperature(C)):40.21858118534789FeedCompressoroutletpressure(Pa)):5100000.0Heateroutlettemperature(C)):215.0Reactoroutlettemperature(C)):231.8500047840716Turbineoutlettemperature(C)):142.0419469...
E
E             ...Full output truncated (4 lines hidden), use '-vv' to show
```

Reply:
@bpaul4 I am working on a better way to test this which avoids the string comparisons. This PR will be updated shrotly.